webservice/components/request_adapter.py

```diff
@@ -17,7 +17,7 @@ def _request(self, method, **kwargs):
             {"auth": self._get_auth(**kwargs), "headers": self._get_headers(**kwargs)}
         )
         request = requests.request(
-            method, self.collection.url.format(**kwargs), **new_kwargs
+            method, self.collection.url.format(**kwargs), **new_kwargs, timeout=None
```

Comment:
Include timeout in the new_kwargs update a couple of lines above, otherwise this will fail is timeout key is specified in when caling _request method